manager/integration/tests/test_node.py

```diff
@@ -2184,3 +2184,49 @@ def test_replica_scheduler_rebuild_restore_is_too_big(client):  # NOQA
     node = common.wait_for_disk_update(client, lht_hostId,
                                        len(update_disks))
     cleanup_host_disk(client, 'vol-small')
+
+
+@pytest.mark.skip(reason="TODO")
+def test_node_eviction():
+    """
+    Test node eviction (assuming this is a 3 nodes cluster)
+
+    1. Disable scheduling on node 3.
+    2. Create volume 1 with 2 replicas.
+    3. Attach volume 1 to node 1 and write some data to it and get
+    checksum 1.
+    4. Set 'Eviction Requested' to 'true' and disable scheduling on node 1.
+    5. Volume 1 should be failed to schedule new replica.
+    6. Set 'Eviction Requested' to 'false' to cancel node 1 eviction and
+    check there should be 1 replica on node 1 and node 2. And remove the
+    unscheduled replica.
+    7. Set 'Eviction Requested' to 'true' on node 1.
+    8. Set 'Replica Node Level Soft Anti-Affinity' to 'true'.
```

Comment:
Should we set 'Replica Node Level Soft Anti-Affinity' to ```True``` before 'Eviction Requested' to ```True``` to make sure replica scheduling doesn't fail on same node.?

Reply:
One reason we set 'Eviction Requested' to 'True' before that is we can see the eviction got suspend and continues.

Reply:
if `Eviction Requested` is set to `True`  before setting `Replica Node Level Soft Anti-Affinity` to `True` 
we should get the same result as step `5.  Volume 1 should be failed to schedule new replica.` and volume should end up with 1 healthy replica one node 2 and one suspended replica since scheduling is disabled on nodes 1 & 3. 
Then after setting `Replica Node Level Soft Anti-Affinity` to `True` replica should scheduled to node 2

@boknowswiki  is that what you mean ?

Reply:
@meldafrawi , step 7, the volume should have 2 healthy replicas, 1 on node 1, 1 on node 2. And 1 suspended replica. Since we only remove the replica when there is healthy replica count > replica spec number, which means we rebuild one first, until it success, we remove the extra one. And after set 'Replica Node Level Soft Anti-Affinity' to 'True', there will be 2 replicas on node 2 (step 9).